### PR TITLE
fix(Views): updated PropTypes to come from prop-types to avoid deprec…

### PR DESCRIPTION
**Pull request by fmal**
…ation warnings for React 15.5+

### CODE REVIEW

File: packages/cerebral/package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
   "homepage": "http://cerebral.github.io/cerebral-website",
   "dependencies": {
     "eventemitter3": "^2.0.2",
-    "function-tree": "^0.5.0-b"
+    "function-tree": "^0.5.0-b",
+    "prop-types": "^15.5.4"
```
Comment (christianalfoni) [inline packages/cerebral/package.json]: This should not be there. React is not a hard dependency to Cerebral, you can also use inferno... and we are keeping options open to use other views as well :)

Comment (henri-hulski) [inline packages/cerebral/package.json]: Maybe instead it should be mentioned in the the docs for react view.

Comment (fmal) [inline packages/cerebral/package.json]: @christianalfoni  good catch. Maybe we could handle it in a similar way as here: https://github.com/paypal/glamorous/commit/14970c1bb59d1408d3066d209dafa356b5ce3173?  Try to require it only when React 15.5 detected and adding a note in the docs about `prop-types` installation?

**Comment by fmal** [comment]
@christianalfoni i removed `prop-types` from dependencies and added React version checking. Users running react > 15.5 will continue to use `React.PropTypes` and there is no extra "overhead" for them. For those who installed react 15.5+, it will require `prop-types` package which is now dependency of React. Since npm3 modules installation is flat it will require no extra package installation from the user 😃

File: packages/cerebral/src/viewFactories/react.js [hunk 2]
```diff
@@ -4,7 +4,23 @@ import StateContainerFactory from './StateContainer'
...
 
-export const Container = ContainerFactory(React)
-export const StateContainer = StateContainerFactory(React)
-export const connect = connectFactory(HocFactory(React))
-export const decorator = decoratorFactory(HocFactory(React))
+let PropTypes
+
+if (React.version.slice(0, 4) === '15.5') {
+  try {
+    PropTypes = require('prop-types')
```
Comment (Guria) [inline packages/cerebral/src/viewFactories/react.js]: it seems that it wouldn't work for react 16.0+ users, right?

Comment (fmal) [inline packages/cerebral/src/viewFactories/react.js]: @guria right, we could adjust the check to be:

```js
const [major, minor] = React.version.split('.');

if ((major === '15' && minor >= 5) || major === '16') {
  ...
}
```
let me try that with `react@16.0.0-alpha.9`

Comment (henri-hulski) [inline packages/cerebral/src/viewFactories/react.js]: Maybe 
```js
if ((major === '15' && minor >= 5) || major >= '16')
```
So it doesn't need to be adjusted on every major react release. Only if it breaks.

**Comment by fmal** [comment]
@Guria @henri-hulski ok it pulls in `prop-types` for all React versions above 15.4, i have tried it with  latest react@16 alpha and everything works correctly

File: packages/cerebral/src/viewFactories/react.js [hunk 3]
```diff
@@ -5,8 +5,9 @@ import HocFactory from './Hoc'
 import connectFactory, { decoratorFactory } from './connect'
 
 let PropTypes
+const [major, minor] = React.version.split('.')
 
-if (React.version.slice(0, 4) === '15.5') {
+if ((major === '15' && minor >= 5) || major >= 16) {
```
Comment (henri-hulski) [inline packages/cerebral/src/viewFactories/react.js]: Maybe remove the quotes also from `'15'` to be consistent. :)

**Comment by Guria** [comment]
Do we really need prop types at all? Maybe just remove it completely?

**Comment by fmal** [comment]
@guria i guess they're needed at least to define `childContexTypes` and `contextTypes`: https://facebook.github.io/react/docs/context.html

**Pull request merged**